nemo_curator/modifiers/llm_pii_modifier.py

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import warnings
+from typing import Dict, List, Optional
+
+from openai import OpenAI
+
+from nemo_curator.modifiers import DocumentModifier
+from nemo_curator.utils.distributed_utils import load_object_on_worker
+from nemo_curator.utils.llm_pii_utils import (
+    JSON_SCHEMA,
+    PII_LABELS,
+    SYSTEM_PROMPT,
+    redact,
+    validate_entity,
+)
+
+__all__ = ["LLMPiiModifier"]
+
+
+class LLMInference:
+    """A class for redacting PII via LLM inference"""
+
+    def __init__(
+        self,
+        base_url: str,
+        api_key: Optional[str] = None,
+        model: str = "meta/llama-3.1-70b-instruct",
+        system_prompt: str = SYSTEM_PROMPT,
+        pii_labels: List[str] = PII_LABELS,
+    ):
+        self.client = OpenAI(base_url=base_url, api_key=api_key)
```

Comment:
I could use our existing `OpenAIClient` from [here](https://github.com/NVIDIA/NeMo-Curator/blob/main/nemo_curator/services/openai_client.py), although I do not see an immediate benefit to this. It seems to just be a wrapper.